Sources/TSCBasic/misc.swift

```diff
@@ -48,9 +48,14 @@ public func getEnvSearchPaths(
     currentWorkingDirectory: AbsolutePath?
 ) -> [AbsolutePath] {
     // Compute search paths from PATH variable.
-    return (pathString ?? "").split(separator: ":").map(String.init).compactMap({ pathString in
+  #if os(Windows)
+    let sep: Character = ";"
+  #else
+    let sep: Character = ":"
+  #endif
+    return (pathString ?? "").split(separator: sep).map(String.init).compactMap({ pathString in
         // If this is an absolute path, we're done.
-        if pathString.first == "/" {
+        if pathString.isAbsolutePath {
```

Comment:
This is only available on `NSString`. I've been thinking about adding some new APIs for such common path operations. Maybe we should add a `Path` struct and expose a static method that can answer this question. It'll also be useful for some other common operations like getting the `extension`. We can use AbsolutePath/RelativePath under the hood.

Reply:
To make sure I understand you, essentially Foundation's [NSPathUtilities](https://github.com/apple/swift-corelibs-foundation/blob/master/Foundation/NSPathUtilities.swift) except they don't go through NSString on Darwin?

Reply:
The problem is that they only go through NSString on Darwin. These properties are unavailable on String. I don't know why corelibs-foundation allows these properties on String. cc @millenomi

Reply:
My impression is that NSString is bridged to String on all platforms. However, on Darwin, it goes through the Objective C NSString while on Linux and Windows it goes through the swift-corelibs-foundation NSString. Is that not correct?